File: twonkfms/fleet_management_system_for_2nk_sacco/report/fms_vehicle/fms_vehicle.js
// Copyright (c) 2023, George Kiirithio Waweru and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["FMS Vehicle"] = {
	"filters": [
		{
			"fieldname": "vehicle",
			"label":__("Vehicle"),
			"fieldtype":"Link",
			"options":"FMS Vehicles",
			"width":100,
			"reqd":0

		},
		{
			"fieldname": "route_name",
			"label":__("Route Name"),
			"fieldtype":"Link",
			"options":"FMS Routes",
			"width":100,
			"reqd":0

		},
		{
			"fieldname": "departure_date",
			"label":__("Departure Date"),
			"fieldtype":"Date",
			"options":"FMS TImetables",
			"width":100,
			"reqd":0

		}
	]
};

File: twonkfms/fleet_management_system_for_2nk_sacco/report/fms_vehicle/fms_vehicle.py
# Copyright (c) 2023, George Kiirithio Waweru and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	columns=get_columns()
	data = get_data(filters)

	return columns, data

def get_data(filters):

	conditions= "1=1"

	if(filters.get("vehicle")):conditions += f" AND vehicle='{filters.get('vehicle')}'"
	if(filters.get("revenue")):conditions += f" AND revenue='{filters.get('revenue')}'"

	SQL = f"""
		SELECT vehicle, SUM(revenue) FROM `tabFMS TImetables` GROUP BY vehicle
		WHERE 'vehicle' = 'kdk200m'
	"""
	
	data=frappe.db.sql(SQL)
	return data

def get_columns():
	return [
		"Vehicle:Link/FMS Vehicles:200",
		"Revenue:Data/FMS TImetables:200"
	]
